pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -695,6 +695,26 @@ type KubeletConfig struct {
 	// FailSwapOn makes the Kubelet fail to start if swap is enabled on the node. (default true).
 	// +optional
 	FailSwapOn *bool `json:"failSwapOn,omitempty" protobuf:"varint,13,opt,name=failSwapOn"`
+	// KubeReserved is the configuration for resources reserved for kubernetes node components.
+	// +optional
+	// Default:
+	//   memory:	1Gi
+	//   cpu:		80m
+	KubeReserved *KubeletConfigKubeReserved `json:"kubeReserved,omitempty" protobuf:"bytes,14,opt,name=kubeReserved"`
+	// SystemReserved is the configuration for resources reserved for system components.
+	// +optional
+	SystemReserved *KubeletConfigSystemReserved `json:"systemReserved,omitempty" protobuf:"bytes,15,opt,name=systemReserved"`
+	// EnforceNodeAllocatable is an array of cgroups whose resource would be enforced by kubelet.
+	// +optional
+	// Default:
+	//   - pods
+	EnforceNodeAllocatable []string `json:"enforceNodeAllocatable,omitempty" protobuf:"bytes,16,opt,name=enforceNodeAllocatable"`
+	// KubeReservedCgroup is the name of the cgroup that kubelet will enforce kube-reserved resource on.
+	// +optional
+	KubeReservedCgroup *string `json:"kubeReservedCgroup,omitempty" protobuf:"bytes,17,opt,name=kubeReservedCgroup"`
+	// SystemReservedCgroup is the name of the cgroup that kubelet will enforce system-reserved resource on.
```

Comment:
// The configured cgroup is not automatically created and thus must exist on the worker node before using this option.